pkg/providers/instance/instance.go

```diff
@@ -279,6 +279,7 @@ func (p *DefaultProvider) filterInstanceTypes(ctx context.Context, instanceTypes
 		instancefilter.CapacityBlockFilter(reqs),
 		instancefilter.ReservedOfferingFilter(reqs),
 		instancefilter.ExoticInstanceTypeFilter(reqs),
+		instancefilter.SpotOfferingFilter(reqs),
```

Comment:
Do we actually need both of these filters? If we just filtered the offending spot offerings out, wouldn't that be sufficient?